### PR TITLE
Nexpose parser bugfixes

### PR DESCRIPTION
**Pull request by damiencarol**
This pull request fixes 2 bugs:
- [x] Fix a bug in tag management
- [x] add missing data for findings without endpoints
- [x] add more unit tests 

Tag works well now:
image

Linked to https://github.com/DefectDojo/django-DefectDojo/issues/3869

### CODE REVIEW

File: dojo/tools/nexpose/parser.py [hunk 1]
```diff
@@ -252,5 +252,52 @@ def get_items(self, tree, vulns, test):
...
+                    false_p=False,
+                    duplicate=False,
+                    out_of_scope=False,
+                    mitigated=None,
+                    dynamic_finding=True)
+                # build references
+                refs = ''
+                for ref in vuln['refs']:
+                    if ref.startswith('CA'):
+                        ref = f" * [{vuln['refs'][ref]}](https://www.cert.org/advisories/{vuln['refs'][ref]}.html)"
```
Comment (kiblik) [inline dojo/tools/nexpose/parser.py:287]: I suppose @damiencarol wanted to write `refs += f"..."`

File: dojo/tools/nexpose/parser.py [hunk 2]
```diff
@@ -252,5 +252,52 @@ def get_items(self, tree, vulns, test):
...
+                    elif ref.startswith('CVE'):
+                        ref = f" * [{vuln['refs'][ref]}](https://cve.mitre.org/cgi-bin/cvename.cgi?name={vuln['refs'][ref]})"
```
Comment (kiblik) [inline dojo/tools/nexpose/parser.py:289]: Same like https://github.com/DefectDojo/django-DefectDojo/pull/3872/files#r576756981

File: dojo/tools/nexpose/parser.py (hunk 1 above)
Comment (damiencarol) [reply dojo/tools/nexpose/parser.py]: not my original code but fixed it :D

File: dojo/tools/nexpose/parser.py (hunk 2 above)
Comment (damiencarol) [reply dojo/tools/nexpose/parser.py]: not my original code but fixed in 4 places. :)

**Comment by dsever** [comment]
@damiencarol to cover all possibility we should add additional possible states:

exception-vulnerable-exploited - Exception suppressed exploited vulnerability
exception-vulnerable-version - Exception suppressed version-checked vulnerability
exception-vulnerable-potential - Exception suppressed potential vulnerability 

https://docs.rapid7.com/nexpose/working-with-report-formats/

**Comment by damiencarol** [comment]
@kiblik thanks for your review! you found a bug that is in the dev branch. :)

**Review by valentijnscholten — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Pull request merged**